packages/api/tests/lambdas/test-cw-sf-event-to-db-records.js

```diff
@@ -170,33 +180,103 @@ test.serial('saveGranulesToDb() does not throw an exception if storeGranulesFrom
   }
 });
 
-test('The cw-sf-event-to-db-records Lambda function creates execution and granule records', async (t) => {
-  const { cumulusMessage, executionModel, granuleModel } = t.context;
+test.serial('savePdrsToDb() saves a PDR record', async (t) => {
+  const { cumulusMessage, pdrModel } = t.context;
+
+  const stateMachineName = randomString();
+  const stateMachineArn = `arn:aws:states:us-east-1:1234:stateMachine:${stateMachineName}`;
+  cumulusMessage.cumulus_meta.state_machine = stateMachineArn;
+
+  const executionName = randomString();
+  cumulusMessage.cumulus_meta.execution_name = executionName;
+  const executionArn = 'https://console.aws.amazon.com/states/home?region=us-east-1#/executions/'
+    + `details/arn:aws:states:us-east-1:1234:execution:${stateMachineName}:${executionName}`;
+
+  const pdr = {
+    name: randomString(),
+    PANSent: false,
+    PANmessage: 'test'
+  };
+  cumulusMessage.payload = {
+    pdr,
+    completed: new Array(4).map(randomString),
+    failed: new Array(2).map(randomString),
+    running: new Array(6).map(randomString)
+  };
+  await savePdrToDb(cumulusMessage);
+
+  const collectionId = (() => {
+    const { name, version } = cumulusMessage.meta.collection;
+    return constructCollectionId(name, version);
+  })();
+
+  const fetchedPdr = await pdrModel.get({ pdrName: pdr.name });
+  const expectedPdr = {
+    pdrName: pdr.name,
+    collectionId,
```

Comment:
Oh, this works because it's an immediately invoked function?

Reply:
Yeah, hence the `();` at the end of that.